### ISSUE TITLE
Stripe PaymentIntents

### ISSUE THREAD

**Issue by Reacye** [tipsi/tipsi-stripe#448]
Are you going to insert support for new Stripe PaymentIntents API?

**Comment by otusweb** [comment]
I've already +1 one this, and if you plan on supporting this, I'm curious about the timeline as the September 2019 deadline is fast approaching.

**Comment by enda-phorest** [comment]
Any update on this?

**Comment by cybergrind** [comment]
@enda-phorest integration doesn't look very hard. But probably tipsi team will implement it only if there are no other options. Probably someone will need it before that and send a PR

**Comment by YJoris** [comment]
Any update ?

**Comment by Ivan-Kachan-Computools** [comment]
Also wondering if there is any update?

**Comment by dziter** [comment]
Hi everyone,

Two days ago, I have sent a message to Stripe to get some update about SCA compliance and integration with or without third-party. 

To resume, the answer is pretty clear: check with Tipsi OR do your own integration. 

@TipsiTeam: Do you plan tu support this feature before September?

« Hello,

This is Mitch with Stripes SCA team stepping in here, I hope this email finds you well. 

Reading through your chat here I see you are using React Native and want to know how to get your integration set up to be SCA compliant. One thing to mention is that most of our third party integration have not completed their SCA updates yet. Partially because not everything is fully moved over yet on our end either. Most are waiting until all of the products and features are done being updated before they change their set up. This puts a lot of companies like you in a bind because you rely on these third parties but they are not ready yet. 

So first and foremost I would reach out to Tipsi about this and find out what their plans are. Then I would using that information plan on what you are going to do to be ready when September rolls around. 

Stripe offers a few easy integration options such as Checkout which now uses payment intents to allow for 3DS and SCA compliance:

https://stripe.com/docs/payments/checkout

But instead of redoing your whole integration first as I said reach out to Tipsi and see what they plan to do and then go from there. If you have further questions I'm always happy to help. 

Best wishes, »

**Comment by cybergrind** [comment]
@dziter our answer remains unchanged:
we will do integration if we don't get any PR from the community. 

Because our major market is the US we don't have tight deadlines for it as other companies who are working in the EU. So there is no ETA from our team, we will do it when we have time for it.

**Comment by enda-phorest** [comment]
Thanks for the update @dziter and @cybergrind.
For more EU focused companies, a PR from the community seems more realistic if you have no ETA at Tipsi.

**Comment by adambutler** [comment]
First of all a huge thanks @cybergrind and the tipsi team for your hard work on this project.

As a UK based product that relies on Stripe we're being forced to move from using the Charge API to the Payment Intents API due to strong customer authentication.

Having read the comments above I totally understand that adding support for Payment Intents isn't a priority for tipsi. However, given the popularity of this library (7,108 weekly downloads at time of writing this) having no clear roadmap for this is quite concerning since literally, our whole business won't be able to operate from September 14th without this migration.

Personally, I'd love to be in a position to contribute a PR but as a super small team of two with no Objective-C or Java experience, this wouldn't really be practical.

Again, to reiterate I get that this _isn't at all_ a tipsi's problem. The library is clearly used by so many companies and products that will be affected. I think we _collectively_ need to devise a clear roadmap and solution to this problem to ensure that all these businesses aren't affected.

Given the breadth of people dependent and that it is currently the _only_ react native library for Stripe I'd be inclined to ask Stripe directly if they are able to contribute. Alternatively, has anyone else got experience in crowdfunding the work needed to add support to this open source project?

I hope you don't mind but I'm copying in some of the top two recent contributors from each of stripe-node, stripe-android and stripe-ios to see if they are able to support (@ob-stripe @remi-stripe @mshafrir-stripe @ksun2-stripe @yuki-stripe @csabol-stripe).

**Comment by trag-stripe** [comment]
@adambutler thanks for raising this, all valid and time-sensitive points.  **+1** to thanking @cybergrind and team for maintaining tipsi for the react native community + Stripe customers. We know this project is a lot to maintain and open source contributions are why react native developers are able to quickly implement payments. 

We're talking internally today on setting a plan and will get back to this thread in short order.

**Comment by cybergrind** [comment]
@trag-stripe API changes itself isn't that hard. 
E2E tests are the issue: 
* #470 blocked by fixing tests and it is blocking all new PRs
* #448 will require new tests. 

So implementing new APIs is 15% of the time, fixing and writing tests - 85%. So it probably will consume at least full week of the one developer and we cannot make this commitment right now.

**Comment by trag-stripe** [comment]
Ok, thanks for the context - makes sense that payment methods, intents, and the required tests need to been seen as all part of the overall work that's required. We had a kick off this morning and are planning to submit relevant PRs to contribute to tipsi-stripe. (Work & progress will be over the next few weeks)

**Comment by adambutler** [comment]
![high five](https://media.giphy.com/media/120jXUxrHF5QJ2/giphy.gif)

👏 Amazing response from both the tipsi and Stripe teams. Thank you.

**Comment by tomrevansecho** [comment]
I have implemented payments intents based off Tipsi for our app. Happy to review PRs when added properly.

**Comment by e-nouri-work** [comment]
@tomrevansecho do you have a fork for what you did please ?

**Comment by tomrevansecho** [comment]
Unfortunately no due as I wasn't sure about the level of PCI compliance and have used native components from Stripe SDK for both iOS and Android. I wrote the bridge for both iOS and Android and this project was a huge help, is some tidy up still to do. 

I would love to contribute back, its finding the time is the hard part.  The Stripe docs were not the easiest to follow but I was lucky I had already done the web implementation, so I didn't require any further backend changes for this. 

The biggest thing to implement is the action for further verification, this was the Android implementation roughly.

`    @ReactMethod
    public void handleCardAction(ReadableMap params, Promise promise) {

        String paymentIntentClientSecret = params.getString(FIELD_PAYMENT_INTENT_CLIENT_SECRET);

        if (paymentIntentClientSecret != null) {

            try {
                final PaymentIntentParams retrievePaymentIntentParams =
                        PaymentIntentParams.createRetrievePaymentIntentParams(
                                paymentIntentClientSecret);

                AsyncTask.execute(() -> {
                    try {
                        // retrieve the PaymentIntent on a background thread
                        final PaymentIntent paymentIntent =
                                mStripe.retrievePaymentIntentSynchronous(
                                        retrievePaymentIntentParams,
                                        BuildConfig.StripeKey);

                        if (paymentIntent != null && paymentIntent.requiresAction()) {

                            Uri redirectUrl = paymentIntent.getRedirectUrl();
                            if (redirectUrl != null) {
                                if (getCurrentActivity() == null) {
                                    promise.reject(
                                            getErrorCode(mErrorCodes, "paymentIntentRetrieveFailed"),
                                            getDescription(mErrorCodes, "paymentIntentRetrieveFailed")
                                    );
                                } else {
                                    mPromise = promise;
                                }

                                getCurrentActivity().startActivity(
                                        new Intent(Intent.ACTION_VIEW, redirectUrl));
                            }
                        } else {
                            promise.resolve(paymentIntent.getId());
                        }
                    } catch (Exception ex) {
                        promise.reject(toErrorCode(ex), ex.getMessage());
                    }
                });
            } catch (Exception ex) {
                promise.reject(toErrorCode(ex), ex.getMessage());
            }
        } else {
            promise.reject(
                    getErrorCode(mErrorCodes, "paymentIntentParameterMissing"),
                    getDescription(mErrorCodes, "paymentIntentParameterMissing")
            );
        }
    }`

and for iOS:

`  @objc func handleCardAction(_ params: Dictionary<String, String>, resolver: @escaping RCTPromiseResolveBlock, rejecter: @escaping RCTPromiseRejectBlock) {
    
    if let paymentIntentClientSecret = params[RNStripeManager.field_payment_intent_client_secret] {
      
      STPAPIClient.shared().retrievePaymentIntent(withClientSecret: paymentIntentClientSecret) { [weak self] (paymentIntent, error) in
        if let paymentIntent = paymentIntent, paymentIntent.status == STPPaymentIntentStatus.requiresAction {
          
          guard let redirectContext = STPRedirectContext(paymentIntent: paymentIntent, completion: { clientSecret, redirectError in
            
            guard redirectError == nil else {
              rejecter(RNStripeManager.error_code, RNStripeManager.error_stripe_unknown, nil);
              return
            }
            
            // Completion block when any redirect flow is done
            STPAPIClient.shared().retrievePaymentIntent(withClientSecret: paymentIntentClientSecret) { confirmPaymentIntent, error in
              if let confirmPaymentIntent = confirmPaymentIntent, confirmPaymentIntent.status == STPPaymentIntentStatus.requiresConfirmation {
                
                resolver(confirmPaymentIntent.stripeId)
                
              } else {
                rejecter(RNStripeManager.error_code, RNStripeManager.error_stripe_unknown, nil);
              }
            }
          }) else {
            
            // Unsupported by Stripe
            rejecter(RNStripeManager.error_code, RNStripeManager.error_stripe_unknown, nil);
            return
          }
          
          if let strongSelf = self {
            strongSelf.redirectContext = redirectContext
            
            // TODO - ... bad practice
            let appDelegate = UIApplication.shared.delegate as! AppDelegate
            redirectContext.startRedirectFlow(from: appDelegate.window.rootViewController!)
          } else {
            rejecter(RNStripeManager.error_code, RNStripeManager.error_stripe_unknown, nil);
          }
        } else {
          rejecter(RNStripeManager.error_code, RNStripeManager.error_stripe_unknown, nil);
        }
      }
    } else {
      rejecter(RNStripeManager.error_code, RNStripeManager.error_stripe_unknown, nil);
    }
  }`

I will try and can get some time but I can't promise anything - happy to review any PRs if anyone makes an earlier start.

**Comment by bamdadd** [comment]
is anyone still working on PaymentIntent for tipsi-stripe?

**Comment by enda-phorest** [comment]
Comment from trag-stripe above indicates that Stripe will kindly do a PR for this issue quite soon.

**Comment by matthiasleitner** [comment]
@trag-stripe can you provide any updates on this?

**Comment by trag-stripe** [comment]
Hi @enda-phorest / @bamdadd / @matthiasleitner - mini update here:

PaymentIntent / SCA support has been in active developement for the last two weeks - point developer on our team is @mindlapse. Context: Beyond PaymentIntents, there were a few pre-existing issues we agreed to resolve around CI Builds and migrating to the latest versions of our Mobile SDKs. 

You can track the key PRs here as they land: https://github.com/tipsi/tipsi-stripe/pulls/mindlapse

**Comment by bamdadd** [comment]
@trag-stripe thank you very much for the mini update. We can't wait for this piece of work and excited that Stripe is getting involved directly !

@mindlapse do you think you can point us to a particular PR for supports on PaymentIntents ?

**Comment by matthiasleitner** [comment]
@trag-stripe thanks for the update! @mindlapse As the deadline for SCA is approaching would be glad if you could share some details about your internal timeline. 

Happy to support on certain PRs if possible.

**Comment by fbartho** [comment]
I'm thrilled to hear that you're working on this @trag-stripe / @mindlapse! Do you recommend using a particular branch? or are they merging right into Tipsi-Master? We're kicking off our backend work, and we want to start adopting tipsi-stripe (we don't currently have stripe in our RN app). So I would love to use the right codebase!

**Comment by bamdadd** [comment]
Please give us an update

**Comment by mindlapse** [comment]
Ideally we'll be bringing the payment intent changes into master.  At the moment we're working through quite a few Travis pipeline/Android emulator/Appium issues to unblock the PRs, and the biggest priority once those are resolved is to bring in the PaymentIntent changes.

@fbartho I know you're interested in a preview build so I'll look to get one set up soon!

**Comment by brooksyd2** [comment]
I'm excited to hear that development on support for the Payment Intents API is in progress. Thanks to all those working on it, particulary @mindlapse for taking point.

The Tipsi libary is a key part of our mobile products, and so as you can imagine i've ben keenly tracking this Issue for a resolution.

Could you please share an upate with where you are, and if possible share the outstanding PR's? I am happy to contribute where possible to getting this over the line given it's importance to us (and many others I'm sure), so let us know if there is anything the community can do to help.

**Comment by mindlapse** [comment]
Hi all,
Really glad to see the interest, there are a couple of gaps where we could use some help.

- [ ] (in progress by @mindlapse) Stabilizing the android CI pipeline 
- [ ] (in progress by @mindlapse) Adding 'AppInfo' changes for Android that identify tipsi-stripe as the client for transactions made through it
- [ ] Adding the same 'AppInfo' changes as above, except for iOS
- [ ] Stabilizing the iOS build pipeline (i.e. so that it doesn't time out from inactivity)
- [ ] Integrating the above into a branch that contains changes for Payment Intents
- [ ] Updating the example app with demo pages using test cards that demonstrate supported payment intent use cases
- [ ] Enhancing the CI pipeline test suite to validate that the test pages operate as expected (and that the existing test suite continues to pass)

Our biggest gap at the moment is with the iOS pipeline - we're seeing timeouts on Travis.  We need someone to look at the iOS pipeline, and to determine if the timeout can be solved on Travis open source build machines, or if we need to move to something like real device testing or a provider that specializes in mobile app testing like SauceLabs.  The GitHub Sponsors program may be a good option to help cover the subscription costs if we go that route.

In terms of PRs, there is #470 which contains changes related to payment intents, these changes will need to go through a review and will need a test suite created and the example app updated.  The changes are really appreciated - I'm looking forward to helping review them.

My focus today is on finishing up the stabilization of the android pipeline, which is nearly ready (i.e. with emulation that doesn't timeout) and I expect to have a PR for those changes soon.

So that's the update!  Thank you for checking in - if you know someone who can volunteer with the iOS pipeline and iOS AppInfo changes, that would be really helpful with moving this forward.

**Comment by hugoh59** [comment]
Thanks for the updates, I'm also following this repository. Depending on the timeline I was considering using a web view inside my RN app in the meantime. Do you guys see any major downside to this approach?

**Comment by shubhamkes** [comment]
Hi @hugoh59  am also doing the same, and manually handling redirection using the next_actions object and as far as i know, It is only method to go with tipsi stripe and performing 3d secure using paymentIntent.
However I am stuck on creating paymentMethod using tipsi-stripe (since we are not a PCI compliant firm, paymentMethod has to be created in client side).
Can anybody help me with that?

**Comment by bamdadd** [comment]
@shubhamkes we are creating paymentMethod on the client side and doing the paymentIntent work on the server side. Here is our fork: https://github.com/b-social/tipsi-stripe
We are still working on it and its not in production yet. But we did some dev testing and it seems to be working on both android and ios

**Comment by JB-CHAUVIN** [comment]
Hello, any updates since the deadline is soon ?

**Comment by mindlapse** [comment]
We're nearing the finish line on the changes and will have a beta branch available soon.  Any volunteers for the beta?  Just curious, what companies are represented here?

**Comment by Bechrissed** [comment]
I'd be interested in participating in the beta branch. My company is CodeCapi and we are developing an app for bobdriver.nl 

Currently not using tipsi at all but planning to switch to tipsi beginning september, to hopefully be in time for sca changes. Our fallback is sending receipts from Stripe with a payment request link, which is not so nice. Have trying to setup intents without tipsi but couldn't make it work because we are currently not using stripe elements.

**Comment by Ivan-Kachan-Computools** [comment]
Would be interested in beta too

**Comment by hugoh59** [comment]
Same here, I'm working on an app called Dogtime Community and would like to roll out a new feature using Stripe PaymentIntent.

**Comment by mindlapse** [comment]
For real-time progress on the code and to chat with those developing this in terms of use cases, I highly recommend joining the discord server for this project.  It's active:
https://discord.gg/8g6ppq2

re: Beta volunteers, great!  We'll have it ready soon and will send out notifications

**Comment by Mark-Hardwick** [comment]
I would also be interested in joining the beta, working on Payments for events etc. for a prop-tech company called District-Technologies. We would like to unblock this feature from being further developed :)

**Comment by RTC1** [comment]
Interested in beta also. 

Ice Leisure Group - ticketing app for our Christmas ice-rinks.

**Comment by Minishlink** [comment]
Good news, thanks @mindlapse!

We're also interested for the beta 👍 

- [BAM](https://www.bam.tech)

**Comment by tgensol** [comment]
Also interested in trying the beta,

[Choose](https://appchoose.io)

Thanks @mindlapse

**Comment by enda-phorest** [comment]
Definitely interested in Beta, we're waiting to release around 2000 apps with this SCA work asap so very keen on testing when available.

Are the below items still requiring some help from the community? 

[ ] Adding the same 'AppInfo' changes as above, except for iOS
[ ] Stabilizing the iOS build pipeline (i.e. so that it doesn't time out from inactivity)
[ ] Integrating the above into a branch that contains changes for Payment Intents
[ ] Updating the example app with demo pages using test cards that demonstrate supported payment intent use cases
[ ] Enhancing the CI pipeline test suite to validate that the test pages operate as expected (and that the existing test suite continues to pass)

**Comment by matthiasleitner** [comment]
@mindlapse  also happy to support beta testing

**Comment by mindlapse** [comment]
@enda-phorest, what would be a huge help right now are new appium tests for the new payment intent demo page on the experimental branch.  A demo page for setupintents is also on the way and will need corresponding coverage there as well

**Comment by Skullflow** [comment]
Interest in for the beta.

Thanks @mindlapse

**Comment by alexsegura** [comment]
Beta-tester here for [CoopCycle](https://github.com/coopcycle/coopcycle-app) ✋

**Comment by JB-CHAUVIN** [comment]
Same, interested for the beta.

**Comment by adambutler** [comment]
@mindlapse Yes please for the beta for [Ordoo](http://github.com/ordoo)

**Comment by dziter** [comment]
@mindlapse : Yes, waiting to try this beta.
Our app with should be release on 01/09 and we still need this feature to be ready.

Thanks a lot for your work.

**Comment by clementinoux** [comment]
Our team is also interested in trying the beta !

[Karamel](https://karamel.cool/) - App to find and book kids activities from newborn to 18 years old

Thanks a lot @mindlapse.

**Comment by joshwarwick** [comment]
We're interested in joining the beta too

[Theodo UK](https://www.theodo.co.uk/)

**Comment by timrc** [comment]
We are also interested in beta https://www.trustedhousesitters.com/

**Comment by e-nouri** [comment]
We are interested in trying the beta, we are a french startup in foodtech.

**Comment by xsv24** [comment]
We are interested in trying the beta we are a start-up in Waste management in the UK.

Thanks for all your hard work!

**Comment by MrKriegler** [comment]
We are also interested in trying the BETA :) we are in Insuratech in the UK.

Thank you so much for your hard work.

**Comment by shinework** [comment]
Hi 🖖 Interested for the beta for @premieroctet 💕

**Comment by MatthieuLab** [comment]
Hi,

Interested in the beta also.
Startup based in Paris, Proptech ind.
Thanks.

**Comment by hullo-rikard** [comment]
Our team is also interested in trying the beta ✌🏼

[Happy Order](https://happyorder.io) - App to order and pay on restaurants

Thanks a lot for your work @mindlapse plus team!

**Comment by liondedan** [comment]
We are interested in trying the beta if possible.
- https://sofarsounds.com

**Comment by fabio571** [comment]
We are interested in beta test too.
[](https://vanillainnovations.com)
Thank you

**Comment by tomrevansecho** [comment]
Would be interested in Beta :)

**Comment by n-sviridenko** [comment]
@mindlapse thanks a lot! [we](https://www.voypost.com/en/)'re also interested in Beta.

**Comment by lucasroca** [comment]
We're a danish electricity supplier startup, very interested in the beta as well.

**Comment by peterjuras** [comment]
Hi,

we're from [ORDA](https://apps.apple.com/de/app/orda/id1218440876?_branch_match_id=688004547271578818), a German startup that allows users to pay in restaurants from their phone. We'd be super interested to join the BETA as well!

Best,
Peter

**Comment by neilkimmett** [comment]
Very interested in the beta.

**Comment by antoninleissen** [comment]
Hi all, 

Very interested in the beta. Work from Brest, Britanny, France. Beable Company. 

Thx for all amazing work !

**Comment by angji** [comment]
Very interested in the beta also.
Thanks for the great work!

**Comment by nmantulenko** [comment]
We are interested in trying the beta if possible.

**Comment by kyytiPetteri** [comment]
Hi! [We](https://www.kyyti.com/) are extremely interested in a beta as well.

**Comment by BernatWozzo** [comment]
Hello! [Courtify's](https://www.courtifyapp.com) team is also interested in a beta!

**Comment by alexander126** [comment]
Hi, GearedApp's team is also interested in the beta!
https://geared.app/

**Comment by getsbydev** [comment]
Hi!
we are [getsby](https://gets.by/) and we are highly interested in trying out the new beta.
Thanks for the awesome work so far!

**Comment by ejkkan** [comment]
Hello!

Definitely want to help test the beta as we will be in need of this feature asap! 
For company: https://www.tiptapp.com/en

Thanks!

**Comment by samuthekid** [comment]
Hi :) We are Redlight Software and would like to test the beta, as we need to use it in some projects.
Thanks

**Comment by marcorm** [comment]
Hi,
i'm interested in the beta for my personal/friends [pojects](https://apps.apple.com/us/developer/marco-marinangeli/id590251776) ;)

**Comment by augustjoergensen** [comment]
Hi

I am working for mover which is Sass company for transportation. www.movertransport.com
We would like access to the beta to service our private customers.

**Comment by rraavv** [comment]
Hi,
i'm interested in the beta for starting a personal project. Thanks!

**Comment by benoitjoseph** [comment]
Hey, I would be interested in trying out the beta as well !

We are a french startup located in Paris

**Comment by jscott22** [comment]
Hey we at Bzzt! would be interested in trying the beta. 
www.bzzt.se

**Comment by Genitivo** [comment]
Hi guys I'm from [Wanderio](www.wanderio.com) and we are interested in using the beta too.

**Comment by albertogallego** [comment]
Hi guys! We're from https://www.thecolvinco.com, a Barcelona based startup, and we want to try this beta as well 👍

**Comment by hanseklund** [comment]
Hi,
We are very interested in trying out the beta too.
https://onceupon.photo 
Startup that enables user to make photo books. 

Already using tipsi-stripe so we are looking for best solution to be SCA compliant.

Thanks!

**Comment by chriswhittleston** [comment]
Hi guys - please sign us (Beautystack - https://www.beautystack.com) up for the beta - thanks.

**Comment by sibe7691** [comment]
Hi, hugely dependent on this fix! Please sign us up! https://appshack.se

**Comment by guiguille** [comment]
Hi all,
Really interested to test it out also, https://rideyego.com

**Comment by howitworks18** [comment]
Hi! Im interested in being part of the Beta as well, please add me to the repo!

**Comment by emilhellblom** [comment]
Hi!
We are also really interested in joining the beta!
Team at [www.yepstr.com](url)

**Comment by emilhellblom** [comment]
Hi!
We are also really interested in joining the beta!
Team at https://yepstr.com

**Comment by adambutler** [comment]
ℹ️ It seems as though SCA requirements may have been delayed for 18 months at least for some of us. Here is what I've been able to piece together:

Based on [a press release from the UK's Financial Services Authority](https://www.fca.org.uk/news/press-releases/fca-agrees-plan-phased-implementation-strong-customer-authentication) it seems that the European Banking Authority recommended a delay of the 14th September PSD2 regulations with regards to strong customer authentication. which has now been confirmed by the UK's FCA.

My understanding though is that it is up to each national authority to announce themselves. This understanding is based on the following qite from [Business Insider](https://www.businessinsider.com/uk-delays-strong-customer-authentication-requirements-2019-8):

> While the European Banking Authority (EBA) did not delay SCA's introduction across the EU, it did note that national authorities could offer "limited additional time" for implementation, which the UK's FCA is now doing. Other nations could follow suit, and since the report that the FCA is reportedly set to endorse notes that there needs to be an EU-wide solution because of cross-border payments, a broader postponement could be on the way.

[Reports started to appear least week](https://www.finextra.com/newsarticle/34229/uk-finance-recommends-18-month-delay-for-new-authentication-rules) suggesting that the UK's Financial Services Authority were considering a delay. These were then followed yesterday with a formal [press release](https://www.fca.org.uk/news/press-releases/fca-agrees-plan-phased-implementation-strong-customer-authentication) confirming the decision. However, at this time I understand that this only affects the UK but it's likely that others will follow.

What I remain unclear on stems from this quote from the FCA press release:

> The FCA will not take enforcement action against firms if they do not meet the relevant requirements for SCA from 14 September 2019 [...]

This suggests to me that banks _may_ still start enforcing SCA rules from 14th September 2019 if they are ready to do so. If this were the case presumably we would have to continue to push to support SCA for risk of payments failing at banks that are have SCA rules in place.

However, in contrast a quote from [this article from the BBC](https://www.bbc.co.uk/news/business-49332023) states that "Retailers are going to be delighted that the FCA is taking a pragmatic approach to enforcement of SCA". This suggests to me that this decision would propagate down to us and that we can expect charges (in the UK at least) to continue to work the same as they have been even beyond 14th September 2019.

Does anyone have any further insight into this?

**Comment by mbovahqo** [comment]
Hey all, my team is very interested in the beta as well.

HqO:  https://www.hqo.co/

**Comment by fbartho** [comment]
@adambutler this is what a stripe rep told me:

>>>There's still a lot of uncertainty around what such a delay would look
like. Although we currently anticipate a gradual enforcement, we still
expect the first banks to start declining payments without two-factor
authentication on September 14th. As a result, we still recommend our
users to finish their migration before then. We have the following
support article that gives more context.

https://support.stripe.com/questions/strong-customer-authentication-sca-enforcement-date

**Comment by howitworks18** [comment]
we're getting a similar response from stripe as well. probably trying to deflect any kind of liability.

**Comment by fbartho** [comment]
@howitworks18 if you click the link in my response, they have updated it, and it references the current status of several countries (unknown) with links to their original phased rollout plans. If you only care about the UK, you might be okay gambling, but currently we have no info about the rest of Europe

**Comment by howitworks18** [comment]
Has anyone that said they were interested in the beta gotten access yet?

**Comment by Minishlink** [comment]
@howitworks18 See https://github.com/tipsi/tipsi-stripe#paymentintents--setupintents (it's in experimental branch)

**Comment by nicopir** [comment]
Hi, I would be really interested for testing the beta version!
I'm working at listminut.be

**Comment by nicolasZZ** [comment]
Hi, me too I would like to begin testing, integration and contribute as I can !
I'm from Dishop company !

**Comment by mbody** [comment]
Hi, I would like to join the beta too ! I'm only working on a prototype for now...
Cheers !

**Comment by pmatsinopoulos** [comment]
Hi! I would like to join. It's the first time we are implementing Stripe in our mobile app. I am from[ HON](https://joinhon.com).

**Comment by mindlapse** [comment]
tipsi-stripe@8.0.0-beta.1 has been published.

This is only a beta - the stable release is planned for Sept 1st.
As a beta participant, please give us feedback about the beta and talk with us on the Discord channel.  Here is the invite link: https://discord.gg/8g6ppq2

**Comment by tremby** [comment]
Is there a summany of any breaking changes between 7.x and 8.x anywhere? Nothing yet in the changelog of the experimental branch.

**Comment by fbartho** [comment]
@tremby -- we haven't collected a full list of the breaking changes. -- Most of the work is new APIs, so the breaking changes should be somewhat small from an API perspective. That said, you will need to change some behaviors if you had already integrated with CardSources/Tokens. This also may include backend changes -- which are obviously out of scope for this library.

**Comment by maartenpeels** [comment]
> `tipsi-stripe@8.0.0-beta.1` has been published, based on the `experimental` branch (thank you @cybergrind for publishing)
> 
> This is only a beta - the stable release is planned for Sept 1st.
> As a beta participant, please give us feedback about the beta and talk with us on the Discord channel. Here is the invite link: https://discord.gg/8g6ppq2

@mikeahirsch

**Comment by gregory-light-it** [comment]
Where is stable release? :(

**Comment by cybergrind** [comment]
@gregory-light-it we need to ensure that it is stable enough if you're looking just for a feature you can use beta release.

**Comment by gregory-light-it** [comment]
That's why I am asking. Cause I want to use it ASAP in production. Currently checking beta on staging

**Comment by cybergrind** [comment]
@gregory-light-it we also have applications that are using it and we need to check that these apps weren't broken by that release or if they were broken how to fix them. Final PR has arrived just on 31 and due to holidays, we weren't able to test it (I really hate to force people to work on holidays).

So if you really want to help with it we need these things:
1) ensure that it is working on your app after upgrade
2) that there all breaking changes have been documented correctly
3) that new functionality works in your app
4) * if you can run library tests, check that they are working on your environment and if they aren't send PR to `experimental` branch with fixes

**Comment by rahulmishra1991** [comment]
Please let us know when stable version release?

**Comment by mindlapse** [comment]
The latest beta, with everything up to this point, was just published by @cybergrind as `tipsi-stripe@8.0.0-beta.4`, _this is effectively the release candidate of the code that we consider to be ready for stable release and you are free to begin integrating it with your app in preparation_.  It's considered to be stable by myself and @fbartho, we've been working through a lot of the changes and we've handed it over as the stable release candidate and the lead maintainers are doing some final checks to ensure that everything is up to par before publishing as stable.

We need your feedback too!  Please join us on the Discord channel and share any findings you may have with beta 4: https://discord.gg/8g6ppq2.  

If you'd like to explore the code (including the CHANGELOG.md) clone the `experimental` branch.  Please note that we support react native 59.x with this new release (not 60).  For android, you will need to update your support libraries to v28.

**Comment by mindlapse** [comment]
In terms of docs, I've extracted this guide to make it easier to get started with using PaymentIntents with tipsi-stripe for SCA compliance:
https://gist.github.com/mindlapse/72139f022d6e620e4f0d59dc50c1797e#file-guide-md

**Comment by saibharani** [comment]
hi
i want to use SCA i am from [saiva](https://saiva.in)

**Comment by fabriziomoscon** [comment]
Thank you so much to @mindlapse and Stripe to chip in with new contribution to this library.

I would like to have access to the BETA releases. My company account is under [Hoxfon](https://www.hoxfon.com/)

**Comment by fbartho** [comment]
@fabriziomoscon and whoever is still waiting on this thread, we have `tipsi-stripe@8.0.0-beta.6` as our latest beta!

**Comment by luisfuertes** [comment]
v `tipsi-stripe@8.0.0-beta.6` not compile in my android project:

```
Searched in the following locations:
       - file:/projectPath/.m2/repository/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.pom
       - file:/projectPath/.m2/repository/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.jar
       - file:/projectPath/MyApp/node_modules/react-native/android/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.pom
       - file:/projectPath/MyApp/node_modules/react-native/android/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.jar
       - file:/projectPath/MyApp/node_modules/jsc-android/dist/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.pom
       - file:/projectPath/MyApp/node_modules/jsc-android/dist/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.jar
       - https://dl.google.com/dl/android/maven2/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.pom
       - https://dl.google.com/dl/android/maven2/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.jar
       - https://jcenter.bintray.com/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.pom
       - https://jcenter.bintray.com/com/github/tipsi/CreditCardEntry/1.5.1/CreditCardEntry-1.5.1.jar
     Required by:
         project :app > project :tipsi-stripe
```

**Comment by rahulmishra1991** [comment]
Please let me know if anyone can help me into below issue

https://stackoverflow.com/questions/57850683/react-native-android-build-failed-with-linking-failed-error

**Comment by palexandrecampos** [comment]
@fbartho iOS build failed with version `tipsi-stripe@8.0.0-beta.6` with errors in TPSStripeManager and TPSCardField. 

Example: 


``` 
▸ Compiling TPSStripeManager.m

❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:115:34: use of undeclared identifier 'STPPaymentMethodTypeCard'

                      @"card": @(STPPaymentMethodTypeCard),
                 ^ ~~~~~~~



❌ /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:116:35: use of undeclared identifier 'STPPaymentMethodTypeiDEAL'

                      @"iDEAL": @(STPPaymentMethodTypeiDEAL),
                                 ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:117:42: use of undeclared identifier 'STPPaymentMethodTypeCardPresent'

                      @"card_present": @(STPPaymentMethodTypeCardPresent),
                                  ^



❌ /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:118:37: use of undeclared identifier 'STPPaymentMethodTypeUnknown'; did you mean 'PKPaymentMethodTypeUnknown'?

                      @"unknown": @(STPPaymentMethodTypeUnknown),
                                         ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:120:20: use of undeclared identifier 'STPPaymentMethodTypeUnknown'; did you mean 'PKPaymentMethodTypeUnknown'?

                   STPPaymentMethodTypeUnknown,
    ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:125:14: use of undeclared identifier 'STPPaymentMethodTypeCard'

        case STPPaymentMethodTypeCard: return @"card";
    ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:126:14: use of undeclared identifier 'STPPaymentMethodTypeiDEAL'

        case STPPaymentMethodTypeiDEAL: return @"iDEAL";
             ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:127:14: use of undeclared identifier 'STPPaymentMethodTypeCardPresent'

        case STPPaymentMethodTypeCardPresent: return @"card_present";
             ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:128:14: use of undeclared identifier 'STPPaymentMethodTypeUnknown'; did you mean 'PKPaymentMethodTypeUnknown'?

        case STPPaymentMethodTypeUnknown: return @"unknown";
             ^



❌ /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:168:20: expected a type

RCT_ENUM_CONVERTER(STPSetupIntentStatus,
    ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:182:43: expected a type

+ (NSString *)STPSetupIntentStatusString:(STPSetupIntentStatus)status {
                   ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:142:23: use of undeclared identifier 'STPPaymentIntentStatusRequiresAction'

                      TPSEntry(requires_action, RequiresAction),
                                          ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:144:23: use of undeclared identifier 'STPPaymentIntentStatusRequiresPaymentMethod'

                      TPSEntry(requires_payment_method, RequiresPaymentMethod),
^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:158:13: use of undeclared identifier 'STPPaymentIntentStatusRequiresAction'

            TPSEntry(requires_action, RequiresAction)
^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:160:13: use of undeclared identifier 'STPPaymentIntentStatusRequiresPaymentMethod'

            TPSEntry(requires_payment_method, RequiresPaymentMethod)
^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:170:23: use of undeclared identifier 'STPSetupIntentStatusUnknown'; did you mean 'STPPaymentIntentStatusUnknown'?

                      TPSEntry(unknown, Unknown),
^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:171:23: use of undeclared identifier 'STPSetupIntentStatusCanceled'; did you mean 'STPPaymentIntentStatusCanceled'?

                      TPSEntry(canceled, Canceled),
    ^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSStripeManager.m:173:23: use of undeclared identifier 'STPSetupIntentStatusRequiresAction'

                      TPSEntry(requires_action, RequiresAction),
    ^



❌  fatal error: too many errors emitted, stopping now [-ferror-limit=]


▸ Compiling TPSError.m
▸ Compiling TPSStripeManager+Constants.m
▸ Compiling TPSCardFieldManager.m
▸ Compiling TPSCardField.m

❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSCardField.m:172:24: expected a type

- (void)setCardParams:(STPPaymentMethodCardParams *)cardParams {
^



❌  /node_modules/tipsi-stripe/ios/TPSStripe/TPSCardField.m:200:38: collection element of type 'NSUInteger' (aka 'unsigned long') is not an Objective-C object

                        @"expMonth": _paymentCardTextField.cardParams.expMonth,
                       ^


error Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening project.xcworkspace
```

Please let me know if anyone can help me! Thanks

**Comment by luisfuertes** [comment]
try with `tipsi-stripe@8.0.0-beta.7`

**Comment by vinoa-team** [comment]
@palexandrecampos ou may also need to update your podfile with a higher version of Stripe if you had a previous version already installed. Updating the npm package won't necessarily update the pod specs. That removed the build errors in my case. On top of moving to tipsi-stripe@8.0.0-beta.7
also

**Comment by palexandrecampos** [comment]
Thanks @vinoa-team, updating version of Stripe in podfile, removed all build errors!

**Comment by fbartho** [comment]
Yeah,
- Xcode 10.3 or later. (Xcode 11 GM not tested yet!)
- stripe-ios 16.0.7 or later (17 not tested by me, but I think I've seen success with it
- react-native 59, 58 tested & happy, (I've heard of mixed results / success with 60, 57 -- but use at your own risk!)

**Comment by palexandrecampos** [comment]
Hey guys! Any solution for this problem when i'm trying to run the project on Android?

```
> Task :app:preDebugBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:preDebugBuild'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve org.bouncycastle:bcprov-jdk15on:1.61.
     Required by:
         project :app > project :tipsi-stripe > com.stripe:stripe-android:10.2.1 > com.stripe:stripe-3ds2-android:1.1.6
      > Could not resolve org.bouncycastle:bcprov-jdk15on:1.61.
         > Could not get resource 'https://jitpack.io/org/bouncycastle/bcprov-jdk15on/1.61/bcprov-jdk15on-1.61.pom'.
            > Could not GET 'https://jitpack.io/org/bouncycastle/bcprov-jdk15on/1.61/bcprov-jdk15on-1.61.pom'. Received status code 522 from server: Origin Connection Time-out
```

**Comment by 7laria** [comment]
in your repositories (android/build.gradle) do you have this?
`   mavenLocal()
        google()
          jcenter()
        // Add jitpack repository (added by tipsi-stripe)
        maven { url "https://jitpack.io" }
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }`

**Comment by palexandrecampos** [comment]
@7laria yes, i've exactly the same in my repositories!

**Comment by 7laria** [comment]
in the same order?

**Comment by palexandrecampos** [comment]
> in the same order?

No, but now it's in the same order, and it works, many thanks 😄

**Comment by cgranetgithub** [comment]
I'd be happy to test the beta for my company www.outbuster.com - I integrated SCA in our website and need to do the same in our (new coming) apps

**Comment by emmanueldd** [comment]
Hi team, 
I would be happy to the test the beta for the mobile app of https://macoiffeuseafro.com

**Comment by marcorm** [comment]
Hi @cgranetgithub, hi @emmanueldd,
you can find the beta version on [npm here](https://www.npmjs.com/package/tipsi-stripe).

Current version is 8.0.0-beta.7.

Hope it helps, bye.

**Comment by alexpchin** [comment]
@mindlapse I believe `8.0.0-beta.7` currently fails to build in Xcode due to: 

<img width="1043" alt="Screenshot 2019-09-18 at 12 19 37" src="https://user-images.githubusercontent.com/40461/65144173-9b374a00-da0e-11e9-95b9-1abd40426f05.png">

**Comment by fbartho** [comment]
@alexpchin you need to upgrade to a newer stripe-ios: 16.0.7 or 17.0.1 or later! Cheers 🍻

**Comment by rafaelpadovani** [comment]
Hi guys, is there anyone getting this error during the build on Android?

> Task :app:preDebugBuild FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:preDebugBuild'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not find com.google.android.gms:play-services-wallet:16.1.0.

**Comment by jeremistadler** [comment]
@rafaelpadovani I solved this by making sure all dependencies have the same version of the library.

I even replaced this line in ./node_modules/tipsi-stripe/android/build.gradle
`com.google.android.gms:play-services-wallet:$googlePlayServicesVersion` to 
`com.google.android.gms:play-services-wallet:16.0.1`

via a post install script using npm package replace-in-file.

**Comment by rafaelpadovani** [comment]
thanks @jeremistadler , it worked by changing the ./node_modules/tipsi-stripe/android/build.gradle file.

**Comment by Donhv** [comment]
error with "tipsi-stripe": "^8.0.0-beta.7" and stripe 16.0.7 
<img width="1117" alt="Screen Shot 2019-09-23 at 10 40 41" src="https://user-images.githubusercontent.com/16864971/65401177-98fb3400-ddf0-11e9-9394-11b277af4023.png">

**Comment by fbartho** [comment]
@Donhv you need a newer Xcode!

**Comment by rahulmishra1991** [comment]
@Donhv this is the fixes for me

Comment out line 232 and 236 in Librares -> TPSStripe.xcodeproj/TPSStripe/TPSStripeManager.m
The go to Libraries -> React.xcodeproj -> Build Settings -> Static Analyzer Analysis Policy and set 'Analyze During Build' to 'No'

none of the above gets saved to git so if you send to another dev / install pods agai nyou may have to re run it so @fbartho was suggesting adding a post install script to check / apply this / send warning

if you have xcode 10.3, you might not need to do the other 2 steps, I just don't have it so required other two steps on 10.1

**Comment by Donhv** [comment]
@rahulmishra1991 thanks for your help
@fbartho I use ApplePay. How to get paymentMethodId with it? I try token.tokenID and card.CardID but it not work

**Comment by fbartho** [comment]
But Xcode 11 might not be recommended, for React-native reasons

**Comment by Mark-Hardwick** [comment]
Any progress on this coming off of Beta? 👀

**Comment by geekazoid-at** [comment]
We are [mybioma.com](url), we want to join your beta program please.

**Comment by SteedMonteiro** [comment]
we are app.studio, we want to join the beta please, thanks.

**Comment by tancredi** [comment]
We are lendable.co.uk, we want to join the beta, please

**Comment by dommy1985** [comment]
we are https://www.urbi.co ,we want to join the beta, please

**Comment by paulux84** [comment]
we are https://www.aroundeen.com,we want to join the beta.
Thanks

**Comment by thomashagstrom** [comment]
We are https://www.crossplatform.se/ and would like to join

**Comment by jezzdk** [comment]
baernholdt.com would like to join the beta

**Comment by thecookieorg** [comment]
mobimaps.io would like to join the beta program

**Comment by tankpower1** [comment]
Hi guys, i'm building an app that requires the use of SCA and would like to sign up for the beta program.

**Comment by tankpower1** [comment]
suera.io would like to join the beta program

**Comment by fbartho** [comment]
For anybody that has come along later, we have Beta builds of 8.0.0 available publicly through NPM, so you can try it there!

**Comment by njho** [comment]
Perfect, thank you. Is there any documentation for the methods to use on the Beta? @fbartho

**Comment by jessmitch42** [comment]
I've been using `@8.0.0-beta` and am wondering if anyone else has come across this error:
`TypeError: _tipsiStripe.default.confirmSetupIntent is not a function`. The examples are super helpful (followed closely so I won't repost my code) but I'm getting this when I try to save a new card.

**Comment by Donhv** [comment]
hi @fbartho how i can folk @8.0.0-beta version 7. i want edit some function.

**Comment by fbartho** [comment]
Hi @Donhv ! The `experimental` branch is where you want to edit!

**Comment by Clem1306** [comment]
Hi everyone ! I have an error with function `stripe.createTokenWithCard(params)`. That's return :

```
Could not Invoke
StripeModule.createTokenWithCard

null
```
I don't find an issue. Anyone had the same error ?

**Comment by hugoh59** [comment]
@jessmitch42 I do, I'm unable to make it work in my app. I have that exact error: 

> TypeError: _tipsiStripe.default.confirmPaymentIntent is not a function

**Comment by jessmitch42** [comment]
@hugoh59 I figured out the problem on my end. 
From the docs:
> Drag the following folder into your project: node_modules/tipsi-stripe/ios/TPSStripe/

I had dragged the wrong folder so it couldn't find the methods. Make sure you're copying `TPSStripe` into your repo's `/ios` folder.

**Comment by VBJeffrey** [comment]
Hi! I would like to join the beta for https://www.bws-networking.com

**Comment by zeallat** [comment]
Hi! I would like to join the beta for https://haulla.com/

**Comment by JClackett** [comment]
Yes!!! ... https://www.noquarter.co

**Comment by gogotanaka** [comment]
Hi, we are https://ticketjam.jp/ would like to join the beta!

**Comment by renaudAmsellem** [comment]
Hey, we are vocaleo (no website yet) and we'd like to join the beta!

**Comment by IjzerenHein** [comment]
Hi, we would also like to join the beta. Our website is https://vicky.one/ and we are a parking startup from the Netherlands.
I personally also have a lot of native iOS/Android experience and can help out isolate issues and add features.
Cheers and keep up the great work!

**Comment by arnaud-zg** [comment]
Hi, we would like to join the beta for https://www.onepark.fr/

**Comment by sajadabasi** [comment]
Hey, I'd like to join the beta!

**Comment by yashwanthm** [comment]
Hi, we'd like to join the beta for https://stayology.com/

**Comment by AideProjet** [comment]
Hi, we are betmanagement.eu and we’d love an early access to the beta.

Thanks !

Théo

**Comment by ssuchanowski** [comment]
do we have any ETA?

**Comment by arnaud-zg** [comment]
> do we have any ETA?

Actually, you can already use `tipsi-stripe@8.0.0-beta.8` and work with payment intents.
https://gist.github.com/mindlapse/72139f022d6e620e4f0d59dc50c1797e

**Comment by ssuchanowski** [comment]
Awesome, thank you!

How production ready is it?

**Comment by arnaud-zg** [comment]
> Awesome, thank you!
> 
> How production ready is it?

I'm not part of the maintainers team, so I don't really know. We actually use payment intents on production environment and don't have any critical issue.

**Comment by matouka-suzuki** [comment]
Hello, we're developing new MaaS app(no website/app yet), we'd like to join the beta please, thanks!

**Comment by Zakyyy** [comment]
I would like to join the beta

**Comment by matouka-suzuki** [comment]
I realized we can use beta. Thanks for the tipsi team for your hard work!
https://github.com/tipsi/tipsi-stripe/issues/448#issuecomment-547084278
https://github.com/tipsi/tipsi-stripe/issues/448#issuecomment-549927428

$ yarn add "https://github.com/tipsi/tipsi-stripe.git#experimental"

**Comment by arnaud-zg** [comment]
> > do we have any ETA?
> 
> Actually, you can already use `tipsi-stripe@8.0.0-beta.8` and work with payment intents.
> https://gist.github.com/mindlapse/72139f022d6e620e4f0d59dc50c1797e

**Comment by rahultrivedy** [comment]
Hi, We are from shiflean.com developing an MVP hybrid app with payment integration. Would love to get early access to the beta, please. Thanks.

**Comment by cleverinventor** [comment]
We are already using tipsi-stripe in our app and would like to start using the beta: https://planning.center 
Thank you.

**Comment by Findiglay** [comment]
Hi, we would also like to start using the beta. We are https://tate.it/.

Grazie!

**Comment by DanGDroid** [comment]
Hi,
we would like to use the beta https://www.farmerdirecto.com/en

**Comment by rahultrivedy** [comment]
**I am trying to send Payment Method as Payment Intent .** 
RN - 0.61 . 
Android.
**I get the error** - 
![Screenshot_20200123-123554](https://user-images.githubusercontent.com/21215823/72965250-c0203d00-3db3-11ea-8592-68afabc93596.jpg)
**The code is:** 
![image](https://user-images.githubusercontent.com/21215823/72965356-05446f00-3db4-11ea-8118-e00617bd693b.png)

**If we use the below code (with PaymentMethod.id) it is working fine but not able to pass the billing details :**

![image (1)](https://user-images.githubusercontent.com/21215823/72965452-3ae95800-3db4-11ea-81de-b2d6d053806b.png)

**Is there any solution where I can send the billing details with Payment Intent to stripe** (compulsory requirement in current project to generate and send invoices to user directly through stripe on successful payment) ?

Any help would be appreciated. Thanks.

**Comment by avaltat** [comment]
Hi I would like to use the beta (working as feelancer)

**Comment by L-Yeiser** [comment]
Hello we, [The Dyrt](https://thedyrt.com/), are implementing a new feature and would love to give the beta a try.

**Comment by mszczech01** [comment]
@rahultrivedy  How were you able to get past the issue mentioned here on android:

> @jessmitch42 I do, I'm unable to make it work in my app. I have that exact error:
> 
> > TypeError: _tipsiStripe.default.confirmPaymentIntent is not a function

**Comment by stonefruit** [comment]
Hey yo. [Kaddra](www.kaddra.com) would like to try the payment intent beta. Thanks so much!

**Comment by taxnair** [comment]
Hello,

Taxfix.de would like to try the beta. Do let us know if this would be possible.

**Comment by Dakojr** [comment]
Hey i want to try the bata for my app .

Thank you ! :)

**Comment by iDuuck** [comment]
I also need to take part in the beta, please.

**Comment by gothraven** [comment]
why paymentRequestWithNativePay doesn't deliver a paymentMethod instead of tokenId and card ? because they can't be used to create a paymentIntent 😢

**Comment by joshgranttc** [comment]
Hi there, we are [tempcover](https://tempcover.com) and we are making a React-Native app at the moment using stripe. Need to be able to  use the 3D secure aspect of stripe, similarly to how we use it on the web. So will need beta please.

**Comment by Raidouane** [comment]
Hey, I would like to try the payment intent beta. Thanks !

**Comment by gothraven** [comment]
I would also like to try the intent beta please, thank you !

**Comment by mats-codes** [comment]
Hey, we're working on a small startup from the north of germany. Integrating Payments into our React-Native App would be the last step until we can launch. Therefore, we'd love to join the beta. Thanks for your great work!

**Comment by cto-pripeljisi** [comment]
Hi, we ([https://www.pripelji.si/](url)) would like to try out the beta in our app.

**Comment by ikhattab** [comment]
I would also like to try the intent beta please, thank you!

**Comment by gothraven** [comment]
guys no one is getting any access yet .. i have no idea if that's normal or its just me !

**Comment by fbartho** [comment]
Hey everyone! Some of this was posted higher up, but to recap:

- You're all in luck! This is a public beta!
- We have published a beta: 8.0.0-beta.9 in npm
- We are working this week on a beta.10

Depending on your exact versions of the Pods/Install method, and your version of Xcode -- beta.9 will not work for you. Until beta.10 comes out, you can install using a GitHub URL pointing at the `experimental` branch, like so:

```json
		"tipsi-stripe": "tipsi/tipsi-stripe#experimental",
```

You can use a specific commit-SHA instead:
```json
		"tipsi-stripe": "tipsi/tipsi-stripe#7772d11360cdc60c419953390b59d11ce6ea614d",
```

My company's app is using that particular commit SHA -- in production since ~September(?) -- but we will update this week as we're upgrading to Xcode 11.4

**Comment by prowalk** [comment]
Would also like to get added to that beta, thanks.

**Comment by thozh** [comment]
> **I am trying to send Payment Method as Payment Intent .**
> RN - 0.61 .
> Android.
> **I get the error** -
> ![Screenshot_20200123-123554](https://user-images.githubusercontent.com/21215823/72965250-c0203d00-3db3-11ea-8592-68afabc93596.jpg)
> **The code is:**
> ![image](https://user-images.githubusercontent.com/21215823/72965356-05446f00-3db4-11ea-8118-e00617bd693b.png)
> 
> **If we use the below code (with PaymentMethod.id) it is working fine but not able to pass the billing details :**
> 
> ![image (1)](https://user-images.githubusercontent.com/21215823/72965452-3ae95800-3db4-11ea-81de-b2d6d053806b.png)
> 
> **Is there any solution where I can send the billing details with Payment Intent to stripe** (compulsory requirement in current project to generate and send invoices to user directly through stripe on successful payment) ?
> 
> Any help would be appreciated. Thanks.

Hey did you find a solution about this ? I'm getting the same error 😞

**Comment by kavindadilshan** [comment]
TypeError: _tipsiStripe.default.authenticatePaymentIntent 

i have error

**Comment by rdjWiss** [comment]
> TypeError: _tipsiStripe.default.authenticatePaymentIntent
> 
> i have error

I have the same error. _tipsiStripe.default.authenticatePaymentIntent is not a function / undefined

Anyone knows how to fix it?

Thank you

**Comment by jcohenho** [comment]
Please add me to the beta! I'll be using it for a personal app project of mine.

**Comment by fbartho** [comment]
@jcohenho please read my comment upthread!: https://github.com/tipsi/tipsi-stripe/issues/448#issuecomment-606175998

@kavindadilshan @thozh please open new issues, as these appear to be specific problems with your installs.

**Comment by jcohenho** [comment]
@fbartho Thank you. Are there docs for the public beta?

**Comment by fbartho** [comment]
All docs and all code are in the `experimental` branch! 

- Docs: https://github.com/tipsi/tipsi-stripe/tree/experimental/website/docs-md
- All Changes from the Experimental branch are viewable: https://github.com/tipsi/tipsi-stripe/pull/532

**Comment by meliodev** [comment]
> confirmPa

I get sometimes this error. But, once I open the file 'node_modules/tipsi-stripe/srce/Stripe.js', it automatically works. I don't understand the cause of this strange behaviour. Has anybody had this issue ?

> 
> 
> @rahultrivedy How were you able to get past the issue mentioned here on android:
> 
> > @jessmitch42 I do, I'm unable to make it work in my app. I have that exact error:
> > > TypeError: _tipsiStripe.default.confirmPaymentIntent is not a function

> confirmPa

I get sometimes this error. But, once I open the file 'node_modules/tipsi-stripe/srce/Stripe.js', it automatically works. I don't understand the cause of this strange behaviour. Has anybody had this issue ?

**Comment by rohitashPrajapati** [comment]
I am also facing a similar issue sometimes. Not knowing what exactly is causing this issue.

_tipsiStripe.default.confirmPaymentIntent is not a function

**Comment by meliodev** [comment]
> confirmPaymentIntent

I think I found a quick fix solution.
On the file you are using that method, just add the following line:

import X from '../../node_modules/tipsi-stripe/src/Stripe'  (relative path to the file Stripe.js)

**Comment by fbartho** [comment]
For everyone fighting with the `_tipsiStripe.default.confirmPaymentIntent is not a function` issue -- in all cases that we have investigated, it was a situation with an incorrect Install & Linkage of the Library and in at least one case a conflict from multiple copies of tipsi-stripe being installed (because it was using npm or something?).

This could be the *wrong version* of React-Native, or not fully completing the manual linkage steps, or a babel or metro configuration issue.

@rohitashPrajapati / @meliodev --
1. Which beta of tipsi-stripe are you using?
2. Which version of React-Native are you using?
3. Which version of babel are you using? & metro?
4. Are you trying to do this with expo? (expo is not supported currently)

**Comment by ifier** [comment]
Any updates?
How can we use 3D secure and PaymentIntents with `tipsi-stripe`?

**Comment by fbartho** [comment]
@ifier have you tried the beta? `tipsi-stripe@8.0.0-beta.10` has had those features for months now -- what appears to be the problem?

(if you read my other comments, you could have been using this since last September)

**Comment by ifier** [comment]
@fbartho 
Thank you. I'm currently on `"tipsi-stripe": "^7.5.3"` and used `createSourceWithParams` to receive `threeDSecure: required` and then start `createSourceWithParams` with type `threeDSecure`. (two `createSourceWithParams` invoked)

I guess I will use Intents soon with `tipsi-stripe@8.0.0-beta.10` or `experimental` cause I need to support Europe as well.

**Comment by fbartho** [comment]
@ifier Sounds good! Yeah, I would highly recommend starting the migration soon then. Intents supports 3DSecure2 worldwide, so you can migrate before you launch in Europe.

Right now `8.0.0-beta.10` is deployed from the head of `experimental`. If any further bug fixes or features are needed, you can monitor the work on the #532 -- but there's nothing I'm currently expecting to merge in the short-term!

**Comment by Clonescody** [comment]
Hey guys, I've been trying to use the beta-10 version to create a paymentMethod and a paymentIntent, but none of the functions are working. Whenever I call `stripe.createPaymentMethod({ card: { token } })` with `token` coming from 

```
const token = await stripe.paymentRequestWithNativePay(
        {
          // requiredBillingAddressFields: ['all'],
          // requiredShippingAddressFields: ['all'],
        },
        [
          {
            label: "Company",
            amount: '14.90',
          },
        ]
      );
```
I get the following error : 
```
JSON value '{
    card =     {
        brand = Visa;
        cardId = "card_1HP9vGCA7AFLBlK1UbM8CB6m";
        country = US;
        dynamicLast4 = 4242;
        expMonth = 12;
        expYear = 2020;
        funding = credit;
        isApplePayCard = 1;
        last4 = 4242;
    };
    created = 1599583439;
    extra =     {
        billingContact = "<null>";
        shippingContact = "<null>";
        shippingMethod = "<null>";
    };
    livemode = 0;
    tokenId = "tok_1HP9vHCA7AFLBlK1MxBFXwzs";
}' of type NSMutableDictionary cannot be converted to NSString
```

I know this is an experimental version, but I was wondering if this feature was still a WIP or if the issue was on my side ? 

Thanks so much guys, any reply appreciated 🙏

**Comment by fbartho** [comment]
@Clonescody -- this would have been better as a separate Github Issue/Help request.

I suspect you're confused around what's returned by `paymentRequestWithNativePay` -- if you look at the payload, and the error message, our tipsi-stripe library on iOS is telling you that it received a hash/dictionary, for a parameter that should have been a string.

What happens if you call `stripe.createPaymentMethod({ card: { token: token.tokenId } })` ?

**Comment by Clonescody** [comment]
Hey @fbartho, I must have tunnel vision a lot to miss that I'm sorry ... 

Thank you so much for your reply anyways :)

**Comment by julietteSparkmate** [comment]
I am currently trying to use Stripe w/ React-Native and found out that Tipsi was my best option, however I can't use confirmPaymentIntent : ` [TypeError: undefined is not a function (near '..._tipsiStripe.default.confirmPaymentIntent...')]`
I have tried to import Stripe from the node_module as suggested by @meliodev and to uninstall (clear cache, rm -rf node_module and check packages version...) as @fbartho suggested... Nothing so far.... Am I missing something ? Additional configurations, checks, or else ? Thanks!

**Comment by fbartho** [comment]
@julietteSparkmate:

- Are you using Expo? (you must eject expo to use tipsi-stripe)
- Are you using version `8.0.0-beta.11`? 
- What versions of react-native are you using? >= 0.59.x required, 0.62 or later recommended.
- What platform iOS,Android, both?

**Comment by julietteSparkmate** [comment]
Thank you @fbartho for this quick answer.
- Tipsi-stripe is working since we can use createTokenWithCard(), however we have expo in our dependencies.
- We are using 8.0.0-beta.11
- React native ~0.61.5
- Both

**Comment by fbartho** [comment]
@julietteSparkmate -- Did you follow the `expo eject` workflow documented here? https://docs.expo.io/bare/customizing/

Have you inspected your `yarn.lock` / `package-lock.json` file to see what version of tipsi-stripe is actually installed?

I ask because`tipsi-stripe@7.5.x` had `createTokenWithCard` but did not have `confirmPaymentIntent` so it's possible that something is accidentally forcing you to have tipsi-stripe on an older version?

Additionally, Expo has a fork of tipsi-stripe that doesn't support `confirmPaymentIntent` so make sure you don't have any conflicting `*payments*` libraries installed. https://docs.expo.io/versions/latest/sdk/payments/

Expo users often have conflicts that were acknowledged to be caused by expo. Since I don't work for Expo (nor for tipsi, nor for Stripe) and don't use expo myself, and expo devs do not actively communicate with tipsi-stripe, it's difficult for me to provide support here.

**Comment by julietteSparkmate** [comment]
Thanks for the advice, we ended up fixing the problem with a simple missing word in the react-native-unimodules installation... It seems to be working perfectly now ! :champagne:

**Comment by trag-stripe** [comment]
**_💳 2020-12-23 - Stripe React Native Updates :atom:_** 

Hi everyone watching this epically long ~saga~ issue,

If you're interested in upcoming releases for {`Stripe + React Native`}, please take a moment and fill out the [Stripe React Native Interest Form](https://app.smartsheet.com/b/form/b1a1053eaac7496eb0ae7892ebd68973). This upcoming year is going to be full of goodies for you and we'll want your feedback, tips, and requests along the way.

As always, thanks for building with our payment APIs!

[@chris_trag](https://twitter.com/chris_trag)
**Stripe Developers**

💬 [twitter/@StripeDev](https://twitter.com/StripeDev)
📺 [youtube/StripeDevelopers](https://youtube.com/stripedevelopers)

**Comment by Joshandrews43** [comment]
@trag-stripe I know this is probably the wrong place for this, but is there any chance we would get a pure JS SDK for react native so people wouldn't need to eject from Expo? Expo offers a ton of great features and speeds up development significantly for a lot of things.

**Comment by cybergrind** [comment]
@trag-stripe that's great news! I put deprecation on the main page and glad that I can end my duty here :D 
Ping me when you will have a public repo link so I could update the description.

@fbartho thank you for all the help!

**Comment by virus2016** [comment]
> @trag-stripe I know this is probably the wrong place for this, but is there any chance we would get a pure JS SDK for react native so people wouldn't need to eject from Expo? Expo offers a ton of great features and speeds up development significantly for a lot of things.

There are some issues here as in Apple Pay and Google pay with Expo. I don't think they will be able to offer native pay using their platform at the moment because of restrictions from both sides. You must declare a merchant for both for builds so their preview tool will not work. 

Apple especially is requiring you to offer Apple Pay inside your app now if you take payment so apps maybe rejected. 

Hopefully someone with more information can correct me if I'm outdated or wrong.

**Comment by trag-stripe** [comment]
Hi folks - checking back in here: Your official Stripe react native library is finally live!

- Public beta announcement: https://twitter.com/StripeDev/status/1391863173926506501
- Public library: https://github.com/stripe/stripe-react-native
- Docs: https://stripe.com/docs/payments/accept-a-payment?platform=react-native

Special thanks for the 500+ devs who gave us feedback during private beta. For now, please try the project out and filing issues if you have questions.

Regarding everyone's questions re: Expo -> we're working with them now that the library is public. More to come, including dozens of supported payment methods, digital wallets, a new payment UI option, and currencies.

@cybergrind - we can now update the repo with these details, thanks!

**Comment by fbartho** [comment]
Closing this ticket, so that new users don't think this project is still active.

See more: 
- https://github.com/tipsi/tipsi-stripe/issues/779
- https://github.com/tipsi/tipsi-stripe/issues/842
